### PR TITLE
Ucsc sql style guide

### PR DESCRIPTION
**Pull request by insuhpak**
### Purpose & Description
Update SQL style guide based on this [discussion](https://github.com/orgs/cdp-ucsc/discussions/71). 

### Additional Notes
A SQLFluff section will be added in the future in a different PR.

### CODE REVIEW

File: cdp-docs/docs/project_resources/sql_style_guide.md
Comment (jtomkins) [inline cdp-docs/docs/project_resources/sql_style_guide.md:1]: Adding a comment here -- 

I like how Brooklyn Data has a TOC at the top of theirs -- could we add one in as well?  I like how you can link to various sections and it less confusing having what seems like two CTE sections --one in Syntax and one in Formatting.

image

**Comment by jtomkins** [comment]
Thank you Insuh -- I really appreciate having the examples in line with the rules **and** keyboard shortcuts! (I need to go back and figure out how I got tabs into my code) 

I left a comment about including a TOC in our version.  

Mentioning in github I was able to preview in rich text mode which showed the changes (including the sections we removed) so I used vscode with a markdown plugin to view it in a cleaner way

**Comment by insuhpak** [comment]
Hi Jen! For the TOC, I think the website may actually generate a sidebar TOC for us based on how the headers (# ## ### ####) were used in the markdown file. 
image
This is probably a good reason for me to learn how to locally build the page to check changes. But I haven't gotten around to downloading node.js to do that. 

Would we be okay with not adding a TOC, checking the website to see if the sidebar TOC is generated, and if not then I can open another PR to add a TOC? 

And for viewing the new version of the style guide, you will have to do what Jen did to view it cleanly or navigate to the PR branch and actual file in Github to view it cleanly!

**Comment by jtomkins** [comment]
>  For the TOC, I think the website may actually generate a sidebar TOC for us based on how the headers (# ## ### ####) were used in the markdown file

Ah k cool! 

> Would we be okay with not adding a TOC, checking the website to see if the sidebar TOC is generated, and if not then I can open another PR to add a TOC?

I am fine with this approach -- approving PR

**Review by jtomkins — APPROVED** [review]
Thank you Insuh!!   I approve this PR. 

I resolved the conversation (which I thought was a comment on that file, still figuring this out...)

**Review by tgraham-ucsc — CHANGES_REQUESTED** [review]
Hi Insuh - nice work on pulling this all together; looks great. But I have some news I wasn't expecting.  After reading the back and forth with Jen, I decided to pull down this branch and build it locally, and to my surprise, it didn't build successfully.  

I googled a bit and found that the `<br>` tag isn't supported but `<br />` apparently is and it did build after I made those changes.  [Reference Link](https://github.com/facebook/docusaurus/discussions/6528)

There was one other change I made on line 330 - For some reason, it was treating the line after the ## as a headline and including it in the TOC, so I added ####

I branched from yours locally and pushed it up if you want to [Compare the Changes](https://github.com/cdp-ucsc/doc/compare/ucsc-sql-style-guide...tg-style-guild-mods?expand=1#diff-cbded6472379efd621e4417d4d875f64a2e9ebfe9cb3b83d04aeae8c09afcc2d) 

It does highlight the need to be able to test the changes before merging; it would be interesting to see how Github Pages would handle the deployment with breaking changes. Would it fail gracefully and leave the existing site unchanged, or would it break it and leave it 1/2 baked?  🤷‍♂️

**Comment by insuhpak** [comment]
Hi Todd thanks for building it locally and catching those issues!

As a learning experience should we actually merge these changes and see how it **does** break the page? Then we can use your branch as the PR that will fix the breaks.

**Comment by tgraham-ucsc** [comment]
> Hi Todd thanks for building it locally and catching those issues!
> 
> As a learning experience should we actually merge these changes and see how it **does** break the page? Then we can use your branch as the PR that will fix the breaks.

Agreed - I can merge it to see what happens, my hope it that it just rejects the build and leaves the existing site in place.

**Review by magants — COMMENTED** [review]
Nice work Insuh!

**Pull request merged**